src/domain-functions.ts

```diff
@@ -212,10 +213,12 @@ function sequence<Fns extends DomainFunction[]>(
   ...fns: Fns
 ): DomainFunction<UnpackAll<Fns>> {
   return function (input: unknown, environment?: unknown) {
-    const [first, ...rest] = fns.map((df) =>
-      A.λ(fromSuccess(applyEnvironment(df, environment))),
+    const dfsAsComposable = fns.map((df) =>
+      A.composable(fromSuccess(applyEnvironment(df, environment))),
     )
-    return dfResultFromcomposable(A.sequence(first, ...rest))(input)
+    return dfResultFromcomposable(
+      A.sequence(...(dfsAsComposable as [Composable])),
```

Comment:
I didn't fully understand why this type cast had to be written as a one element tuple since `sequence` will take any non-empty list. But this type cast should have no effect outside the function so I'm happy to leave it as is.

Reply:
Check if my previous comment applies here